socorro/cron/jobs/first_signatures.py

```diff
@@ -0,0 +1,204 @@
+    app_description = 'Using SuperSearch to log when signatures first appeared'
+    app_version = '0.1'
+
+    required_config = Namespace()
+    required_config.add_option(
+        'window_seconds',
+        default=60,  # seconds
+        doc='Number of seconds to search for signature facets back'
+    )
+    required_config.add_option(
+        'sensitive_to_errors',
+        # default=True,
+        default=False,
+        doc=(
+            'If there are, for example, missing indexes in ES you can get errors '
+            "in supersearch doing queries across dates we don't have indices for. "
+            'If this is the case, halt this crontabber app.'
+        )
+    )
+    required_config.add_option(
+        'historic_days',
+        # The number 26 comes from the default in index_cleaner which is used
+        # as a retention policy for cleaning elasticsearch.
+        default=26 * 7,  # days
+        doc=(
+            'When looking for the oldest crash possible (by a specific signature) '
+            'go back this many days'
+        )
+    )
+    required_config.add_option(
+        'elasticsearch_class',
+        default=ElasticsearchConfig,
+    )
+
+    def run(self, date):
+        # First get a list of all unique signatures, in the given timespan
+        all_fields = SuperSearchFields(config=self.config).get()
+        api = SuperSearch(config=self.config)
+        assert isinstance(date, datetime.datetime), type(date)
+        start_date = date - datetime.timedelta(seconds=self.config.window_seconds)
+        params = {
+            'date': [
+                '>={}'.format(start_date.isoformat()),
+                '<{}'.format(date.isoformat()),
+            ],
+            '_facets': 'signature',
+            '_facets_size': 1000,
+            '_results_number': 0,
+            '_fields': all_fields,
+        }
+        results = api.get(**params)
+        signatures = [
+            x['term'] for x in results['facets']['signature']
+        ]
+        self.config.logger.info(
+            'Found %s unique signatures between %s and %s',
+            len(signatures),
+            start_date,
+            date,
+        )
+        if not signatures:
+            self.config.logger.info(
+                'Exit early because no signature facets'
+            )
+            return
+
+        # Now to figure out which ones we've never seen before
+        signature_first_date_api = SignatureFirstDate(config=self.config)
+        results = signature_first_date_api.get(signatures=signatures)
+        new_signatures = set(signatures) - set(results['hits'])
+        self.config.logger.info('%s new signatures', len(new_signatures))
+
+        # Now, for each of these new signatures we have to find the oldest
+        # crash that has it and write down its date and buildid.
+        historic_date = date - datetime.timedelta(
+            days=self.config.historic_days
+        )
+        for signature in new_signatures:
```

Comment:
It's sad that you have to iterate over the signatures. Elasticsearch would allow you to do that with just a single query, using a ``min`` aggregation ([doc](https://www.elastic.co/guide/en/elasticsearch/reference/1.4/search-aggregations-metrics-min-aggregation.html)). Unfortunately SuperSearch doesn't support that. Maybe that feature would be worth adding?

Reply:
There are so few of these. New signatures in 24 hours are rare. If we have, god forbid, 5 new signatures in a day we'd have to do 2*5 supersearch queries (+1 for the cardinality facet search). It's nothing for the crontabber to poke around with. It has time.

Reply:
Yeah, I agree. It's just a shame that it isn't easy to solve this in a more elegant way. :)

Reply:
Once this is up and running I can't imagine there being that many *new* signatures *per hour*.